app/bundles/FormBundle/Controller/ResultController.php

```diff
@@ -392,6 +394,218 @@ public function deleteAction(Request $request)
+            \assert($model instanceof SubmissionModel);
+
+            /** @var Submission|null $submission */
+            $submission = $model->getEntity($objectId);
+
+            if (null === $submission) {
+                $flashes[] = [
+                    'type'    => 'error',
+                    'msg'     => 'mautic.form.error.notfound',
+                    'msgVars' => ['%id%' => $objectId],
+                ];
+            } elseif (!$this->security->hasEntityAccess('form:forms:editown', 'form:forms:editother', $submission->getCreatedBy())) {
+                return $this->accessDenied();
+            } else {
+                $domain = $this->getDomainFromSubmission($submission);
+
+                if ($domain) {
+                    $formatted     = '*@'.$domain;
+                    $domains       = $this->coreParametersHelper->get('do_not_submit_emails', []);
+                    $isNewDomain   = !in_array($formatted, $domains, true);
+
+                    if ($isNewDomain) {
+                        $domains[] = $formatted;
+                        $configurator->mergeParameters(['do_not_submit_emails' => $domains]);
+                        $configurator->write();
+                    }
+
+                    $this->enableDonotSubmitValidationOnUnconfiguredForms();
+
+                    $flashes[] = [
+                        'type'    => 'notice',
+                        'msg'     => 'mautic.form.result.markspam.success',
+                        'msgVars' => ['%domain%' => $domain],
+                    ];
+                } else {
+                    $flashes[] = [
+                        'type' => 'error',
+                        'msg'  => 'mautic.form.result.markspam.error',
+                    ];
+                }
+            }
+        }
+
+        $viewParameters = [
+            'objectId' => $formId,
+            'page'     => $page,
+        ];
+
+        return $this->postActionRedirect(
+            [
+                'returnUrl'       => $this->generateUrl('mautic_form_results', $viewParameters),
+                'viewParameters'  => $viewParameters,
+                'contentTemplate' => 'Mautic\\FormBundle\\Controller\\ResultController::indexAction',
+                'passthroughVars' => [
+                    'mauticContent' => 'formresult',
+                ],
+                'flashes' => $flashes,
+            ]
+        );
+    }
+
+    public function batchMarkSpamAction(Request $request, Configurator $configurator): \Symfony\Component\HttpFoundation\RedirectResponse|Response
+    {
+        $formId  = $this->getFormIdFromRequest();
+        $session = $request->getSession();
+        $page    = $session->get('mautic.formresult.'.$formId.'.page', 1);
+        $flashes = [];
+
+        if (Request::METHOD_POST === $request->getMethod()) {
+            $model = $this->getModel('form.submission');
+            \assert($model instanceof SubmissionModel);
```

Comment:
DI
```suggestion
    public function batchMarkSpamAction(Request $request, Configurator $configurator, SubmissionModel $model): \Symfony\Component\HttpFoundation\RedirectResponse|Response
    {
        $formId  = $this->getFormIdFromRequest();
        $page    = $request->getSession()->get('mautic.formresult.'.$formId.'.page', 1);
        $flashes = [];

        if (Request::METHOD_POST === $request->getMethod()) {
```

There is also a lot of duplication between the the batch and single action. Perhaps it could be extracted to a private method and be used in both?